crypto/keyring/record.go

```diff
@@ -0,0 +1,132 @@
+package keyring
+
+import (
+	"errors"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+var ErrPrivKeyExtr = errors.New("private key extraction works only for Local")
+
+func NewRecord(name string, pk cryptotypes.PubKey, item isRecord_Item) (*Record, error) {
```

Comment:
There are a lof of `New*` functions here, and it's a bit confusing.

I propose only four public functions in this file:
```go
func NewLocalRecord(name, pub cryptotypes.PubKey, priv cryptotypes.PrivKey) (*Record, error)
func NewLedgerRecord(name, pub cryptotypes.PubKey, path hd.BIP44Params) (*Record, error)
func NewMultiRecord(name, pub cryptotypes.PubKey) (*Record, error)
func NewOfflineRecord(name, pub cryptotypes.PubKey) (*Record, error)
```

and all other functions (if needed) are private